qiskit_nature/circuit/library/initial_states/hartree_fock.py

```diff
@@ -39,7 +40,14 @@ def __init__(
             num_particles: The number of particles as a tuple storing the number of alpha- and
                            beta-spin electrons in the first and second number, respectively.
             qubit_converter: a QubitConverter instance.
+
+        Raises:
+            ValueError: If qubit_converter contains BravyiKitaevSuperFastMapper.
         """
+        if isinstance(qubit_converter.mapper, BravyiKitaevSuperFastMapper):
+            raise ValueError(
+                "Unsupported mapper in qubit_converter: ", type(qubit_converter.mapper), "."
```

Comment:
Accordingly the phrasing could be a little bit updated. We could for example point to #436 where one can read up on why this mapper is not supported.